wolfcrypt/src/port/aria/aria-cryptocb.c

```diff
@@ -39,6 +39,7 @@ size and a key size of 128, 192, or 256 bits.
 #include <wolfssl/ssl.h>
 #include <wolfssl/wolfcrypt/asn_public.h>
 #include <wolfssl/wolfcrypt/port/aria/aria-cryptocb.h>
+#include <wolfssl/wolfcrypt/ecc.h>
```

Comment:
Same question here as in `aria-crypt.c`

Reply:
The `ecc_set_type` used in [cryptocb.c](https://github.com/wolfSSL/wolfssl/blob/827287000c7f075eff0708f0b8839c778f10fd67/wolfcrypt/src/port/aria/aria-cryptocb.c#L182) is [defined in ecc.h](https://github.com/wolfSSL/wolfssl/blob/827287000c7f075eff0708f0b8839c778f10fd67/wolfssl/wolfcrypt/ecc.h#L289) 

I suspect a setting in the regular `make` causes it to be already included.

Omitting that `#include` for `cmake` gives these build errors:

```
[  1%] Building C object CMakeFiles/wolfssl.dir/wolfcrypt/src/port/aria/aria-cryptocb.c.o
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-cryptocb.c: In function ‘wc_AriaSign’:
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-cryptocb.c:183:11: error: unknown type name ‘ecc_set_type’
  183 |     const ecc_set_type* dp;
      |           ^~~~~~~~~~~~
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-cryptocb.c:217:12: error: invalid use of incomplete typedef ‘ecc_key’
  217 |     if (key->dp != NULL) {
      |            ^~
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-cryptocb.c:218:17: error: invalid use of incomplete typedef ‘ecc_key’
  218 |         dp = key->dp;
      |                 ^~
[... snip ... ]

many other similar errors....
```

Reply:
The chain of includes should go wolfssl/wolfcrypt/port/aria/aria-cryptocb.h -> wolfssl/wolfcrypt/cryptocb.h -> wolfssl/wolfcrypt/ecc.h.

Possible reasons for it to go awry would be WOLF_CRYPTO_CB not defined or if this is on windows then maybe some interaction with the WOLFSSL_ECC_CURVE_STATIC define which gets set on Windows and slightly changes the ecc_set_type struct?

Reply:
Hi @JacobBarthelmeh - do you prefer that I include `cryptocb.h` _instead_ of `ecc.h` in my [proposed changes](https://github.com/wolfSSL/wolfssl/pull/6730/files)?

Doing so does not work by default. I see a ton of errors. 

I tried to force ECC on by adding this to the beginning of the `aria-crypt.c` and `aria-cryptcb.c` files:

```
#undef HAVE_ECC
#define HAVE_ECC
```

For example:
```
wolfcrypt/src/port/aria/aria-cryptocb.c:261:11: error: unknown type name ‘ecc_set_type’
  261 |     const ecc_set_type* dp;
      |           ^~~~~~~~~~~~
wolfcrypt/src/port/aria/aria-cryptocb.c:297:12: error: invalid use of incomplete typedef ‘ecc_key’
  297 |     if (key->dp != NULL) {
      |            ^~
wolfcrypt/src/port/aria/aria-cryptocb.c:298:17: error: invalid use of incomplete typedef ‘ecc_key’
  298 |         dp = key->dp;
      |                 ^~
wolfcrypt/src/port/aria/aria-cryptocb.c:301:41: error: invalid use of incomplete typedef ‘ecc_key’
  301 |         dp = wc_ecc_get_curve_params(key->idx);
      |                                         ^~
wolfcrypt/src/port/aria/aria-cryptocb.c:304:11: error: request for member ‘id’ in something not a structure or union
  304 |     if (dp->id == ECC_SECP256R1) {
```

I could try and chase this one to see what directives need to be turned on, but it seems considerably more straightforward to simply include the needed `wolfcrypt/ecc.h` file.

Note with the additional ARIA includes and the changes in https://github.com/wolfSSL/wolfssl/pull/6730, the dependency on having OpenSSL enabled has been removed.

Let me know which route your prefer. Thanks.